src/app/shared/directives/digit-only.directive.spec.ts

```diff
@@ -0,0 +1,16 @@
+import { DigitOnlyDirective } from './digit-only.directive';
+
+describe('DigitOnlyDirective', () => {
+  it('should create an instance', () => {
+    const directive = new DigitOnlyDirective();
+    expect(directive).toBeTruthy();
+  });
+  it('should listen to KeyboardEvent', () => {
```

Comment:
Rename test to `should call prevent default on non numeric key press` and you can elaborate on your event object like this
const event: KeyboardEvent = {
 key: a,
 preventDefault: () => {}
}

So you can check for both numeric and and non-numeric parsing in the test. So likely one more test for numeric